File: output/ChatGPT/avatar/Python/Java/atcoder_AGC010_B.java
int n = Integer.parseInt(scanner.nextLine());
String[] inputArray = scanner.nextLine().split(" ");
int[] a = new int[n];
for(int i=0; i<n; i++) {
    a[i] = Integer.parseInt(inputArray[i]);
}
if (Arrays.stream(a).sum() % (n * (n + 1) / 2) != 0) {
    System.out.println("NO");
} else {
    int k = Arrays.stream(a).sum() / (n * (n + 1) / 2);
    int[] extendedArray = new int[n+1];
    System.arraycopy(a, 0, extendedArray, 0, n);
    extendedArray[n] = a[0];
    int sum1 = 0;
    int sum2 = 0;
    for (int i = 1; i <= n; i++) {
        sum1 += (k - extendedArray[i] + extendedArray[i-1]) % n;
        sum2 += (k - extendedArray[i] + extendedArray[i-1]) / n >= 0 ? 1 : 0;
    }
    if (sum1 != 0 || sum2 != n) {
        System.out.println("NO");
    } else {
        System.out.println("YES");
    }
}

File: output/ChatGPT/avatar/Java/Python/atcoder_AGC010_B.py
def main():
    N = int(input())
    A = list(map(int, input().split()))

    if possible(N, A):
        print("YES")
    else:
        print("NO")

def possible(N, A):
    sum = 0
    for i in range(N):
        sum += A[i]
    
    NS = N * (N + 1) // 2
    if sum % NS != 0:
        return False
    
    K = sum // NS
    for i in range(N):
        j = N - 1 if i == 0 else i - 1
        d = K - (A[i] - A[j])
        if d < 0 or d % N != 0:
            return False
    
    return True

if __name__ == "__main__":
    main()